### PR TITLE
Separate CPU and GPU tests for Transforms torchscript test

### PR DESCRIPTION
**Pull request by mthrok**
Currently torchscript tests for Transforms module runs GPU tests optionally when GPU is available.
This makes it hard for understand what test is ran/skipped/success/fail.

This PR separates CPU test and GPU test and make them explicit in test report.

Relates to #515 

[Screen Shot 2020-04-07 at 13 25 46](https://travis-ci.org/github/pytorch/audio/jobs/672182174)

### CODE REVIEW

File: test/test_torchscript_consistency.py [hunk 1]
```diff
@@ -304,85 +304,75 @@ def test_perf_biquad_filtering(self):
...
-
-        py_method = py_method.cuda()
-        jit_method = torch.jit.script(py_method)
+class _TransformsTestMixin:
+    """Implements test for Transforms that are performed for different devices"""
+    device = None
 
-        py_out = py_method(tensor)
-        jit_out = jit_method(tensor)
+    def _assert_conssitency(self, transform, tensor):
```
Comment (vincentqb) [inline test/test_torchscript_consistency.py:321]: nit: typo "consistency"

File: test/test_torchscript_consistency.py [hunk 2]
```diff
@@ -304,85 +304,75 @@ def test_perf_biquad_filtering(self):
...
+        _assert_transforms_consistency(transform, tensor, self.device)
 
-        torch.testing.assert_allclose(jit_out, py_out)
-
-
-class TestTransforms(unittest.TestCase):
     def test_Spectrogram(self):
         tensor = torch.rand((1, 1000))
-        _test_script_module(torchaudio.transforms.Spectrogram, tensor)
+        self._assert_conssitency(T.Spectrogram(), tensor)
```
Comment (vincentqb) [inline test/test_torchscript_consistency.py:326]: Were you trying to reduce the use of classes/methods in each test?

File: test/test_torchscript_consistency.py [hunk 3]
```diff
@@ -304,85 +304,75 @@ def test_perf_biquad_filtering(self):
         _test_torchscript_functional(F.lfilter, waveform, a, b)
 
 
-RUN_CUDA = torch.cuda.is_available()
-print("Run test with cuda:", RUN_CUDA)
+def _assert_transforms_consistency(transform, tensor, device):
```
Comment (vincentqb) [inline test/test_torchscript_consistency.py:307]: Does this make the interface different for functionals?

File: test/test_torchscript_consistency.py [hunk 4]
```diff
@@ -391,23 +381,32 @@ def test_Fade(self):
...
+        self._assert_conssitency(T.Vol(1.1), waveform)
+
+
+class TestTransformsCPU(_TransformsTestMixin, unittest.TestCase):
+    """Test suite for Transforms module on CPU"""
+    device = torch.device('cpu')
+
+
+@unittest.skipIf(not torch.cuda.is_available(), 'CUDA not available')
+class TestTransformsCUDA(_TransformsTestMixin, unittest.TestCase):
```
Comment (vincentqb) [inline test/test_torchscript_consistency.py:407]: Overall, this looks reasonable to me. What would be an alternative way?

File: test/test_torchscript_consistency.py (hunk 3 above)
Comment (mthrok) [reply test/test_torchscript_consistency.py:307]: Yes, functionals do not have `to` method, so Tensor arguments have to be moved to the target device manually.

File: test/test_torchscript_consistency.py (hunk 2 above)
Comment (mthrok) [reply test/test_torchscript_consistency.py:326]: Yes, that's why the actual implementation is placed outside of the class and this helper method is a syntax sugar to pass `self.device` automatically. This way, even if we want to get rid of class structure, it's easy.

File: test/test_torchscript_consistency.py (hunk 4 above)
Comment (mthrok) [reply test/test_torchscript_consistency.py:407]: Actually, I do not come up with an alternative that is not hacky and as good as this one.
PyTorch's test dynamically generates this by manipulating global variable scope, but that's basically same thing as this implementation., yet more messy and it does not bring any additional benefit for us.

File: test/test_torchscript_consistency.py (hunk 3 above)
Comment (vincentqb) [reply test/test_torchscript_consistency.py:304]: Would there be a way to keep the two implementations closer?

Comment (mthrok) [reply test/test_torchscript_consistency.py:304]: Refer to my [WIP](https://github.com/mthrok/audio/commit/c736e21746653c1132654d80adb83e71dc96d757#diff-04757c68c2f80adf8dac82791cd9d5beR13). To me this is very close.

**Pull request merged**